src/Processors/QueryPlan/ReadFromRemote.cpp

```diff
@@ -472,7 +492,15 @@ void ReadFromRemote::addLazyPipe(Pipes & pipes, const ClusterProxy::SelectStream
                 max_remote_delay = std::max(try_result.delay, max_remote_delay);
         }
 
-        if (try_results.empty() || local_delay < max_remote_delay)
+        const time_t local_delay = replicated_storage->getAbsoluteDelay();
```

Comment:
let's add a comment that a race can happen here but it should be really rare

Reply:
If the current replica is the most stale one, we will never decide to read from it.